From d5d515a9f491edb5fa7ae8a64fa1a4f6c1474154 Mon Sep 17 00:00:00 2001
From: Rashed Abdel-Tawab <rashed@linux.com>
Date: Fri, 21 Sep 2018 18:24:01 -0700
Subject: [PATCH 14/37] lineage: Move qcom pathmap setting into "BoardConfig"

Since we're making pathmaps set soong namespaces now, the project
pathmaps need to be set at BoardConfig reading time. Move them to
the right place.

Example soong namespace list:
PRODUCT_SOONG_NAMESPACES= hardware/ril hardware/qcom/audio-caf/msm8998
    hardware/qcom/display-caf/msm8998 hardware/qcom/media-caf/msm8998
    hardware/qcom/camera device/essential/mata/gps hardware/qcom/sensors
    vendor/qcom/opensource/location vendor/qcom/opensource/dataservices
    hardware/qcom/power hardware/qcom/thermal hardware/qcom/vr
    hardware/qcom/wlan-caf hardware/qcom/bt-caf

Change-Id: I718a4d042b555802f769fe931b91e0338e6aee37
---
 build/core/config.mk         |  3 --
 build/core/qcom_target.mk    | 58 ------------------------------------
 build/core/qcom_utils.mk     | 20 +++++++++++++
 config/BoardConfigLineage.mk |  2 --
 config/BoardConfigQcom.mk    | 45 ++++++++++++++++++++++++----
 5 files changed, 60 insertions(+), 68 deletions(-)
 delete mode 100644 build/core/qcom_target.mk

diff --git a/build/core/config.mk b/build/core/config.mk
index 71a73d0e..2f758c97 100644
--- a/build/core/config.mk
+++ b/build/core/config.mk
@@ -24,6 +24,3 @@ BUILD_RRO_SYSTEM_PACKAGE := $(TOPDIR)vendor/lineage/build/core/system_rro.mk
 
 # Rules for MTK targets
 include $(TOPDIR)vendor/lineage/build/core/mtk_target.mk
-
-# Rules for QCOM targets
-include $(TOPDIR)vendor/lineage/build/core/qcom_target.mk
diff --git a/build/core/qcom_target.mk b/build/core/qcom_target.mk
deleted file mode 100644
index 1efd516c..00000000
--- a/build/core/qcom_target.mk
+++ /dev/null
@@ -1,58 +0,0 @@
-# Bring in Qualcomm helper macros
-include vendor/lineage/build/core/qcom_utils.mk
-
-define wlan-set-path-variant
-$(call project-set-path-variant,wlan,TARGET_WLAN_VARIANT,hardware/qcom/$(1))
-endef
-define bt-vendor-set-path-variant
-$(call project-set-path-variant,bt-vendor,TARGET_BT_VENDOR_VARIANT,hardware/qcom/$(1))
-endef
-
-# Set device-specific HALs into project pathmap
-define set-device-specific-path
-$(if $(USE_DEVICE_SPECIFIC_$(1)), \
-    $(if $(DEVICE_SPECIFIC_$(1)_PATH), \
-        $(eval path := $(DEVICE_SPECIFIC_$(1)_PATH)), \
-        $(eval path := $(TARGET_DEVICE_DIR)/$(2))), \
-    $(eval path := $(3))) \
-$(call project-set-path,qcom-$(2),$(strip $(path)))
-endef
-
-ifeq ($(BOARD_USES_QCOM_HARDWARE),true)
-
-$(call set-device-specific-path,AUDIO,audio,hardware/qcom/audio-caf/$(QCOM_HARDWARE_VARIANT))
-$(call set-device-specific-path,DISPLAY,display,hardware/qcom/display-caf/$(QCOM_HARDWARE_VARIANT))
-$(call set-device-specific-path,MEDIA,media,hardware/qcom/media-caf/$(QCOM_HARDWARE_VARIANT))
-
-$(call set-device-specific-path,CAMERA,camera,hardware/qcom/camera)
-$(call set-device-specific-path,DATA_IPA_CFG_MGR,data-ipa-cfg-mgr,vendor/qcom/opensource/data-ipa-cfg-mgr)
-$(call set-device-specific-path,GPS,gps,hardware/qcom/gps)
-$(call set-device-specific-path,SENSORS,sensors,hardware/qcom/sensors)
-$(call set-device-specific-path,LOC_API,loc-api,vendor/qcom/opensource/location)
-$(call set-device-specific-path,DATASERVICES,dataservices,vendor/qcom/opensource/dataservices)
-$(call set-device-specific-path,POWER,power,hardware/qcom/power)
-$(call set-device-specific-path,THERMAL,thermal,hardware/qcom/thermal)
-$(call set-device-specific-path,VR,vr,hardware/qcom/vr)
-
-$(call wlan-set-path-variant,wlan-caf)
-$(call bt-vendor-set-path-variant,bt-caf)
-
-PRODUCT_CFI_INCLUDE_PATHS += \
-    hardware/qcom/wlan-caf/qcwcn/wpa_supplicant_8_lib
-else
-
-$(call project-set-path,qcom-audio,hardware/qcom/audio/default)
-$(call project-set-path,qcom-display,hardware/qcom/display/$(TARGET_BOARD_PLATFORM))
-$(call project-set-path,qcom-media,hardware/qcom/media/$(TARGET_BOARD_PLATFORM))
-
-$(call project-set-path,qcom-camera,hardware/qcom/camera)
-$(call project-set-path,qcom-data-ipa-cfg-mgr,hardware/qcom/data/ipacfg-mgr)
-$(call project-set-path,qcom-gps,hardware/qcom/gps)
-$(call project-set-path,qcom-sensors,hardware/qcom/sensors)
-$(call project-set-path,qcom-loc-api,vendor/qcom/opensource/location)
-$(call project-set-path,qcom-dataservices,$(TARGET_DEVICE_DIR)/dataservices)
-
-$(call wlan-set-path-variant,wlan)
-$(call bt-vendor-set-path-variant,bt)
-
-endif
diff --git a/build/core/qcom_utils.mk b/build/core/qcom_utils.mk
index f36ecaa0..497b875e 100755
--- a/build/core/qcom_utils.mk
+++ b/build/core/qcom_utils.mk
@@ -246,3 +246,23 @@ $(strip \
   $(if $(strip $(acn)),true,) \
 )
 endef
+
+# Set wlan variant into project pathmap
+define wlan-set-path-variant
+$(call project-set-path-variant,wlan,TARGET_WLAN_VARIANT,hardware/qcom/$(1))
+endef
+
+# Set bluetooth variant into project pathmap
+define bt-vendor-set-path-variant
+$(call project-set-path-variant,bt-vendor,TARGET_BT_VENDOR_VARIANT,hardware/qcom/$(1))
+endef
+
+# Set device-specific HALs into project pathmap
+define set-device-specific-path
+$(if $(USE_DEVICE_SPECIFIC_$(1)), \
+    $(if $(DEVICE_SPECIFIC_$(1)_PATH), \
+        $(eval path := $(DEVICE_SPECIFIC_$(1)_PATH)), \
+        $(eval path := $(TARGET_DEVICE_DIR)/$(2))), \
+    $(eval path := $(3))) \
+$(call project-set-path,qcom-$(2),$(strip $(path)))
+endef
diff --git a/config/BoardConfigLineage.mk b/config/BoardConfigLineage.mk
index 7bbe01e8..223f7d22 100644
--- a/config/BoardConfigLineage.mk
+++ b/config/BoardConfigLineage.mk
@@ -5,9 +5,7 @@ endif
 
 include vendor/lineage/config/BoardConfigKernel.mk
 
-ifeq ($(BOARD_USES_QCOM_HARDWARE),true)
 include vendor/lineage/config/BoardConfigQcom.mk
-endif
 
 include vendor/lineage/config/BoardConfigSoong.mk
 
diff --git a/config/BoardConfigQcom.mk b/config/BoardConfigQcom.mk
index 344ced84..fce78a4d 100644
--- a/config/BoardConfigQcom.mk
+++ b/config/BoardConfigQcom.mk
@@ -1,6 +1,8 @@
 # Bring in Qualcomm helper macros
 include vendor/lineage/build/core/qcom_utils.mk
 
+ifeq ($(BOARD_USES_QCOM_HARDWARE),true)
+
 B_FAMILY := msm8226 msm8610 msm8974
 B64_FAMILY := msm8992 msm8994
 BR_FAMILY := msm8909 msm8916
@@ -88,12 +90,45 @@ endif
 endif
 endif
 
-PRODUCT_SOONG_NAMESPACES += \
-    hardware/qcom/audio-caf/$(QCOM_HARDWARE_VARIANT) \
-    hardware/qcom/display-caf/$(QCOM_HARDWARE_VARIANT) \
-    hardware/qcom/media-caf/$(QCOM_HARDWARE_VARIANT)
-
 # QCOM HW crypto
 ifeq ($(TARGET_HW_DISK_ENCRYPTION),true)
     TARGET_CRYPTFS_HW_PATH ?= vendor/qcom/opensource/cryptfs_hw
 endif
+
+$(call set-device-specific-path,AUDIO,audio,hardware/qcom/audio-caf/$(QCOM_HARDWARE_VARIANT))
+$(call set-device-specific-path,DISPLAY,display,hardware/qcom/display-caf/$(QCOM_HARDWARE_VARIANT))
+$(call set-device-specific-path,MEDIA,media,hardware/qcom/media-caf/$(QCOM_HARDWARE_VARIANT))
+
+$(call set-device-specific-path,CAMERA,camera,hardware/qcom/camera)
+$(call set-device-specific-path,DATA_IPA_CFG_MGR,data-ipa-cfg-mgr,vendor/qcom/opensource/data-ipa-cfg-mgr)
+$(call set-device-specific-path,GPS,gps,hardware/qcom/gps)
+$(call set-device-specific-path,SENSORS,sensors,hardware/qcom/sensors)
+$(call set-device-specific-path,LOC_API,loc-api,vendor/qcom/opensource/location)
+$(call set-device-specific-path,DATASERVICES,dataservices,vendor/qcom/opensource/dataservices)
+$(call set-device-specific-path,POWER,power,hardware/qcom/power)
+$(call set-device-specific-path,THERMAL,thermal,hardware/qcom/thermal)
+$(call set-device-specific-path,VR,vr,hardware/qcom/vr)
+
+$(call wlan-set-path-variant,wlan-caf)
+$(call bt-vendor-set-path-variant,bt-caf)
+
+PRODUCT_CFI_INCLUDE_PATHS += \
+    hardware/qcom/wlan-caf/qcwcn/wpa_supplicant_8_lib
+
+else
+
+$(call project-set-path,qcom-audio,hardware/qcom/audio/default)
+$(call project-set-path,qcom-display,hardware/qcom/display/$(TARGET_BOARD_PLATFORM))
+$(call project-set-path,qcom-media,hardware/qcom/media/$(TARGET_BOARD_PLATFORM))
+
+$(call project-set-path,qcom-camera,hardware/qcom/camera)
+$(call project-set-path,qcom-data-ipa-cfg-mgr,hardware/qcom/data/ipacfg-mgr)
+$(call project-set-path,qcom-gps,hardware/qcom/gps)
+$(call project-set-path,qcom-sensors,hardware/qcom/sensors)
+$(call project-set-path,qcom-loc-api,vendor/qcom/opensource/location)
+$(call project-set-path,qcom-dataservices,$(TARGET_DEVICE_DIR)/dataservices)
+
+$(call wlan-set-path-variant,wlan)
+$(call bt-vendor-set-path-variant,bt)
+
+endif
-- 
2.17.1

